src/internal_utils.jl

```diff
@@ -0,0 +1,25 @@
+"""
+This file contains utility functions that are not specific to the
+library, but are used throughout.
+"""
+
+@generated function fieldnames_equal(::Type{T1}, ::Type{T2}) where {T1,T2}
+    # Needs to be a generated function as might not be inlined
+    return Base.fieldnames(T1) == Base.fieldnames(T2)
```

Comment:
Does this line make sense for `SymbolicDimensions`? It looks like `SymbolicDimensions` directly overloads `static_fieldnames` in order to indicate its fields, and this wouldn't be reflected in `fieldnames`? https://github.com/SymbolicML/DynamicQuantities.jl/blob/09a43cc4d9dd55c596d41a852d65342bdbe4b49f/src/symbolic_dimensions.jl#L42

Separately, it doesn't seem right to make `static_fieldnames` the API to the extent that it needs to be overloaded by `SymbolicDimensions`. Maybe better to make the API something like `dimension_names`, which defaults to `static_fieldnames`? I'll leave it to your discretion what (if anything) needs to be addressed here rather than in a seprate PR.

Reply:
Good catch!

Reply:
Going to add this in a future PR